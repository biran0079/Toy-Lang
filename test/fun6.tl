fun main() {
  fun f() {
    print("in main");
  }
  fun g() {
    fun f() {
      print("in g");
    }
    f();
  }
  f();
  g();
  f();
  fun f() {
    print ("in main 2");
  }
}

main();

fun f() {
  print("out main");
}

f();
main();
f();

