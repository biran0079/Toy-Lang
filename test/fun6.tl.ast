(stmts (fun (id main) (ids) (stmts (fun (id f) (ids) (stmts (tail_call (id print) (exps (string "in main"))))) (fun (id g) (ids) (stmts (fun (id f) (ids) (stmts (tail_call (id print) (exps (string "in g"))))) (tail_call (id f) (exps)))) (call (id f) (exps)) (call (id g) (exps)) (call (id f) (exps)) (fun (id f) (ids) (stmts (tail_call (id print) (exps (string "in main 2"))))))) (call (id main) (exps)) (fun (id f) (ids) (stmts (tail_call (id print) (exps (string "out main"))))) (call (id f) (exps)) (call (id main) (exps)) (call (id f) (exps)))
