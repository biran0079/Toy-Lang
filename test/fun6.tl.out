in main
in g
in main
out main
in main
in g
in main
out main
